modules/text2vec-cohere/vectorizer/objects.go

```diff
@@ -59,22 +89,44 @@ func (v *Vectorizer) Object(ctx context.Context, object *models.Object, cfg modu
 
 func (v *Vectorizer) object(ctx context.Context, object *models.Object, cfg moduletools.ClassConfig,
 ) ([]float32, error) {
-	icheck := NewClassSettings(cfg)
+	icheck := ent.NewClassSettings(cfg)
 	text := v.objectVectorizer.Texts(ctx, object, icheck)
 
-	res, err := v.client.Vectorize(ctx, []string{text}, ent.VectorizationConfig{
-		Model:   icheck.Model(),
-		BaseURL: icheck.BaseURL(),
-	})
+	res, _, err := v.client.Vectorize(ctx, []string{text}, cfg)
 	if err != nil {
 		return nil, err
 	}
-	if len(res.Vectors) == 0 {
+	if len(res.Vector) == 0 {
 		return nil, fmt.Errorf("no vectors generated")
 	}
 
-	if len(res.Vectors) > 1 {
-		return libvectorizer.CombineVectors(res.Vectors), nil
+	if len(res.Vector) > 1 {
+		return libvectorizer.CombineVectors(res.Vector), nil
 	}
-	return res.Vectors[0], nil
+	return res.Vector[0], nil
+}
+
+func (v *Vectorizer) ObjectBatch(ctx context.Context, objects []*models.Object, skipObject []bool, cfg moduletools.ClassConfig,
+) ([][]float32, map[int]error) {
+	texts := make([]string, len(objects))
+	tokenCounts := make([]int, len(objects))
+	icheck := ent.NewClassSettings(cfg)
+
+	// prepare input for vectorizer, and send it to the queue. Prepare here to avoid work in the queue-worker
+	skipAll := true
+	for i := range objects {
+		if skipObject[i] {
+			continue
+		}
+		skipAll = false
+		text := v.objectVectorizer.Texts(ctx, objects[i], icheck)
```

Comment:
Why are we vectorizing one by one? Cohere's API is supports batch (https://docs.cohere.com/reference/embed)

Reply:
This just creates a single text out of the different properties, nothing is send to cohere